confgenerator/confmerger.go

```diff
@@ -47,8 +48,8 @@ func MergeConfFiles(ctx context.Context, userConfPath string, builtInConfStructs
 
 	// Ensure the merged config struct fields are valid.
 	v := newValidator()
-	if err := v.Struct(result); err != nil {
-		panic(err)
+	if err := v.StructCtx(ctx, result); err != nil {
+		log.Fatalf("merged config failed to validate: %v", err)
```

Comment:
Why change this from a panic to a log? It should be impossible for the merged config to be invalid if the user config is valid (and we've already validated the user config by this point).

Reply:
`log.Fatalf` still panics, it just prints the panic message in a nicer format.